presubmit_tests/single-instance-on-gcp.sh

```diff
@@ -0,0 +1,141 @@
+if [[ -z "${project_id}" ]]; then
+  echo "Could not get the project ID"
+  exit 1
+fi
+gcs_source="${gcs_bucket}/${toolkit_zip_file_name}"
+deployment_id="projects/${project_id}/locations/${location}/deployments/${deployment_name}"
+
+cleanup() {
+  if [[ -n "${tail_pid}" ]]; then
+    kill "${tail_pid}"
+  fi
+  echo "Cleaning up: deleting ${gcs_source} GCS object and ${deployment_id} Infra Manager deployment..."
+  if gcloud infra-manager deployments describe "${deployment_id}" >/dev/null 2>&1; then
+    gcloud --quiet infra-manager deployments delete "${deployment_id}" 
+  fi
+  if gcloud storage objects describe "${gcs_source}" >/dev/null 2>&1; then
+    gcloud --quiet storage rm "${gcs_source}"
+  fi
+}
+
+trap cleanup SIGINT SIGTERM EXIT
+
+echo "Zipping CWD into /tmp/${toolkit_zip_file_name} and uploading to ${gcs_bucket}/..."
+zip -r /tmp/"${toolkit_zip_file_name}" . -x ".git*" -x ".terraform*" -x "terraform*" -x OWNERS > /dev/null
+gcloud storage cp /tmp/"${toolkit_zip_file_name}" "${gcs_bucket}/"
+
+sed -i "s|@deployment_name@|$deployment_name|g" "${tfvars_file}"
+sed -i "s|@gcs_source@|$gcs_source|g" "${tfvars_file}"
+sed -i "s|@instance_name@|$instance_name|g" "${tfvars_file}"
+
+echo "Applying Infra Manager deployment: ${deployment_id}"
+gcloud infra-manager deployments apply "${deployment_id}" \
+  --service-account="projects/${project_id}/serviceAccounts/infra-manager-deployer@${project_id}.iam.gserviceaccount.com" \
+  --local-source="./terraform" \
+  --inputs-file="${tfvars_file}" || exit 1
+
+echo "Resources created by the ${deployment_name} deployment"
+gcloud infra-manager resources list \
+--deployment="${deployment_name}" \
+--location="${location}" \
+--revision=r-0 || exit 1
+
+# Extract the id of the control node resource
+# The format is: projects/<project>/zones/<zone>/instances/control-node-<random-suffix>
+control_node_resource_id="$(gcloud infra-manager resources list \
+--deployment="${deployment_name}" \
+--location="${location}" \
+--revision=r-0 \
+--filter='terraformInfo.type=google_compute_instance' \
+--format='value(terraformInfo.id)')"
+if [[ -z "${control_node_resource_id}" ]]; then
+  echo "Could not retrieve control node's resource ID."
+  exit 1
+fi
+echo "Control node resource ID: ${control_node_resource_id}"
+
+# Get the instance ID from the instance name
+control_node_instance_id="$(gcloud compute instances describe "${control_node_resource_id}" --format="value(id)")"
+if [[ -z "${control_node_instance_id}" ]]; then
+  echo "Could not get control node's instance ID."
+  exit 1
+fi
+echo "Control node instance ID: ${control_node_instance_id}"
+
+# Stream logs from the startup script execution to stdout in the background
+# https://cloud.google.com/logging/docs/reference/tools/gcloud-logging#install_live_tailing
+gcloud --quiet components install beta || exit 1
+export CLOUDSDK_PYTHON_SITEPACKAGES=1
+gcloud beta logging tail \
```

Comment:
PYTHONWARNINGS="ignore" gcloud beta logging tail \

to avoid those known warnings, plus a comment to say why perhaps